dataplane/linux/vxlan_mgr.go

```diff
@@ -59,47 +74,83 @@ type vxlanManager struct {
 	ipSetMetadata     ipsets.IPSetMetadata
 	externalNodeCIDRs []string
 	vtepsDirty        bool
+	nlHandle          vxlanDataplane
+	dpConfig          Config
+	parentProtocol    int
+	// Used so that we can shim the parent route table for the tests
+	parentRTConstruct func(interfacePrefixes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
+		deviceRouteSourceAddress net.IP, deviceRouteProtocol int, removeExternalRoutes bool) routeTable
 }
 
 func newVXLANManager(
 	ipsetsDataplane ipsetsDataplane,
-	maxIPSetSize int,
-	hostname string,
-	routeTable routeTable,
+	rt routeTable,
+	deviceName string,
+	dpConfig Config,
+) *vxlanManager {
+	nlHandle, _ := netlink.NewHandle()
+
+	return newVXLANManagerWithShims(
+		ipsetsDataplane,
+		rt,
+		deviceName,
+		dpConfig,
+		nlHandle,
+		func(interfacePrefixes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
+			deviceRouteSourceAddress net.IP, deviceRouteProtocol int, removeExternalRoutes bool) routeTable {
```

Comment:
Do we need the extra layer of indirection here by wrapping in another function? Couldn't we just do this?

```
parentRTConstruct: routetable.New,
```

Reply:
unfortunately not, and the reasoning being is because golang sucks with interfacing. The constructor returns a routetable.RouteTable instance, which technically does not much the func definition of returning a routeTable instance. Can't wait for go generics